api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/ListOfPatientsWithMdsEvaluationCohortQueries.java

```diff
@@ -1121,30 +1123,33 @@ public DataDefinition getSecondViralLoadResult() {
     map.put("1065", hivMetadata.getYesConcept().getConceptId());
 
     String query =
-        "SELECT     second_vl.patient_id, "
-            + "          IF(oo.value_numeric IS NOT NULL, oo.value_numeric, IF(oo.value_coded = 165331, CONCAT('MENOR QUE ',oo.comments), oo.value_coded)) AS second_vl_result  "
-            + "FROM       patient second_vl "
-            + "INNER JOIN encounter ee "
-            + "ON         ee.patient_id = second_vl.patient_id "
-            + "INNER JOIN obs oo "
-            + "ON         oo.encounter_id = ee.encounter_id "
-            + "INNER JOIN ( "
-            + getSecondVlResultDateQuery()
-            + "  ) second_vl_date "
-            + " ON second_vl_date.patient_id = second_vl.patient_id "
-            + "WHERE second_vl.voided = 0 "
-            + "AND ee.voided = 0 "
-            + "AND oo.voided = 0 "
-            + "AND ee.encounter_datetime = second_vl_date.second_vl_date "
-            + "AND      ee.encounter_type = ${6} "
-            + "AND        ee.location_id = :location "
+        "SELECT     p.patient_id, "
+            + "     IF(o.value_numeric IS NOT NULL, o.value_numeric, IF(o.value_coded = 165331, CONCAT('MENOR QUE ',o.comments), o.value_coded)) AS first_vl_result  "
```

Comment:
This if clause `IF(o.value_numeric IS NOT NULL, o.value_numeric, ...` might not always return the expected result, due to the` o.value_numeric IS NOT NULL` verification that can return true but for a different concept.
Ex: if a patient is included on the definition with qualitative VL (`o.concept_id = ${1305} and o.value_coded IS NOT NULL`), when the IF clause is triggered, if the patient has another **concept with value_numeric not null**, the **o.value_numeric** (which is not the criteria including him), can return true. 

So, you have to make sure that the returned results are exactly from the criteria that is including the patient on the definition.